xdsl/ir.py

```diff
@@ -389,11 +392,17 @@ def print_parameters(self, printer: Printer) -> None:
         """Print the attribute parameters."""
         printer.print_paramattr_parameters(self.parameters)
 
+    def _verify(self):
```

Comment:
If we want to copy `Operation` (which we should), `verify` should be the one calling the `irdl` check,
and be the one calling `verify_`.
Also, this should be `verify_` and not `_verify`? Or if we make that change, we should do it in `Operation` as well.

Reply:
I don't really know why we call it `verify_` in Operation tbh, the trailing underscore is just ugly. It should be `verify` or `custom_verify`. If we do make the changes to make them consistent, I'm not sure that this PR is the ideal place to do it.

Reply:
This is an artifact of me implementing it before this was a real project.
We should just change the name of `verify_` to a better name, and return an error if `verify_` is defined.
But yes we can in another PR.